r/src/main/scala/org/apache/zeppelin/rinterpreter/KnitRInterpreter.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.rinterpreter
+
+
+// TODO:  Capture the knitr progress bar
+
+import java.util._
+
+import org.apache.zeppelin.interpreter.{InterpreterContext, InterpreterResult}
+import org.apache.zeppelin.rinterpreter.rscala.RException
+
+class KnitRInterpreter(property: Properties) extends RInterpreter(property) {
+
+
+  logger.info("Starting KnitR")
+
+  def this() = {
+    this(new Properties())
+  }
+
+  override def open: Unit = {
+    logger.debug("Opening")
+    rContext.synchronized {
+      super.open
+      logger.debug("Knitr open, initial commands")
+      rContext.testRPackage("knitr", true, true, "Without knitr, the knitr interpreter cannot run.")
+      rContext.eval(
+        """opts_knit$set(out.format = 'html',
+          |results='asis',
+          |progress = FALSE,
+          |self.contained = TRUE,
+          |verbose = FALSE,
+          |comment = NA,
+          |tidy = FALSE)
+          |
+          |.z.show.googleVis <- function(widget) print(widget, tag = 'chart')
+          |.z.show.rCharts <- function(widget) widget$show('inline', include_assets=TRUE, cdn=TRUE)
+          | """.stripMargin)
+    }
+    logger.info("KnitR:  Finished initial commands")
+  }
+
+  def interpret(st: String, context: InterpreterContext): InterpreterResult = try { {
```

Comment:
None - thanks for catching that. At one point I used IntelliJ to reformat the code for Apache styles and it left a bunch of those artifacts. Thought I caught them all.

> On Aug 13, 2015, at 3:45 AM, Martin Grigorov <notifications@github.com> wrote:
> 
> In r/src/main/scala/org/apache/zeppelin/rinterpreter/KnitRInterpreter.scala:
> 
> > +        """opts_knit$set(out.format = 'html',
> > +          |results='asis',
> > +          |progress = FALSE,
> > +          |self.contained = TRUE,
> > +          |verbose = FALSE,
> > +          |comment = NA,
> > +          |tidy = FALSE)
> > +          |
> > +          |.z.show.googleVis <- function(widget) print(widget, tag = 'chart')
> > +          |.z.show.rCharts <- function(widget) widget$show('inline', include_assets=TRUE, cdn=TRUE)
> > +          | """.stripMargin)
> > +    }
> > +    logger.info("KnitR:  Finished initial commands")
> > +  }
> > +
> > +  def interpret(st: String, context: InterpreterContext): InterpreterResult = try { {
> What is the need for double { ?
> 
> —
> Reply to this email directly or view it on GitHub.
>